examples/config.c

```diff
@@ -0,0 +1,304 @@
+			oc->name = strdup(output_name);
+			oc->transform = WL_OUTPUT_TRANSFORM_NORMAL;
+			wl_list_insert(&config->outputs, &oc->link);
+		}
+
+		if (strcmp(name, "x") == 0) {
+			oc->x = strtol(value, NULL, 10);
+		} else if (strcmp(name, "y") == 0) {
+			oc->y = strtol(value, NULL, 10);
+		} else if (strcmp(name, "rotate") == 0) {
+			if (strcmp(value, "90") == 0) {
+				oc->transform = WL_OUTPUT_TRANSFORM_90;
+			} else if (strcmp(value, "180") == 0) {
+				oc->transform = WL_OUTPUT_TRANSFORM_180;
+			} else if (strcmp(value, "270") == 0) {
+				oc->transform = WL_OUTPUT_TRANSFORM_270;
+			} else if (strcmp(value, "flipped") == 0) {
+				oc->transform = WL_OUTPUT_TRANSFORM_FLIPPED;
+			} else if (strcmp(value, "flipped-90") == 0) {
+				oc->transform = WL_OUTPUT_TRANSFORM_FLIPPED_90;
+			} else if (strcmp(value, "flipped-180") == 0) {
+				oc->transform = WL_OUTPUT_TRANSFORM_FLIPPED_180;
+			} else if (strcmp(value, "flipped-270") == 0) {
+				oc->transform = WL_OUTPUT_TRANSFORM_FLIPPED_270;
+			} else {
+				wlr_log(L_ERROR, "got unknown transform value: %s", value);
+			}
+		}
+	} else if (strcmp(section, "cursor") == 0) {
+		if (strcmp(name, "map-to-output") == 0) {
+			config->cursor.mapped_output = strdup(value);
+		} else if (strcmp(name, "geometry") == 0) {
+			if (config->cursor.mapped_geo) {
+				free(config->cursor.mapped_geo);
+			}
+			config->cursor.mapped_geo = parse_geometry(value);
+		} else {
+			wlr_log(L_ERROR, "got unknown cursor config: %s", name);
+		}
+	} else if (strncmp(device_prefix, section, strlen(device_prefix)) == 0) {
+		const char *device_name = section + strlen(device_prefix);
+		struct device_config *dc;
+		bool found = false;
+
+		wl_list_for_each(dc, &config->devices, link) {
+			if (strcmp(dc->name, device_name) == 0) {
+				found = true;
+				break;
+			}
+		}
+
+		if (!found) {
+			dc = calloc(1, sizeof(struct device_config));
+			dc->name = strdup(device_name);
+			wl_list_insert(&config->devices, &dc->link);
+		}
+
+		if (strcmp(name, "map-to-output") == 0) {
+			if (dc->mapped_output) {
+				free(dc->mapped_output);
+			}
+			dc->mapped_output = strdup(value);
+		} else if (strcmp(name, "geometry") == 0) {
+			if (dc->mapped_geo) {
+				free(dc->mapped_geo);
+			}
+			dc->mapped_geo = parse_geometry(value);
+		} else {
+			wlr_log(L_ERROR, "got unknown device config: %s", name);
+		}
+	} else {
+		wlr_log(L_ERROR, "got unknown config section: %s", section);
+	}
+
+	return 1;
+}
+
+struct example_config *parse_args(int argc, char *argv[]) {
+	struct example_config *config = calloc(1, sizeof(struct example_config));
+	wl_list_init(&config->outputs);
+	wl_list_init(&config->devices);
+
+	int c;
+	while ((c = getopt(argc, argv, "C:h")) != -1) {
+		switch (c) {
+		case 'C':
+			config->config_path = strdup(optarg);
+			break;
+		case 'h':
+		case '?':
+			usage(argv[0], c != 'h');
+		}
+	}
+
+	if (!config->config_path) {
+		// get the config path from the current directory
+		char cwd[1024];
```

Comment:
Probably want to make that 4k as that is the max on most system, or use MAXPATHLEN from sys/param.h